packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -689,8 +689,8 @@ class Analytics implements IAnalytics {
 
   // eslint-disable-next-line class-methods-use-this
   getSessionId(): Nullable<number> {
-    this.userSessionManager?.refreshSession();
-    return state.session.sessionInfo.value?.id ?? null;
+    const sessionId = this.userSessionManager?.getSessionId();
```

Comment:
This seems incorrect. If the session expires, the caller will not get the refreshed value.

Reply:
We need to have a different strategy for marking the session start status.

Reply:
I am creating another task for changing the strategy for marking the session start status.

Reply:
https://linear.app/rudderstack/issue/SDK-576/change-the-strategy-to-denote-the-mark-of-session-start